src/main/java/org/mule/extension/api/metadata/MetadataResolver.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.metadata;
+
+import org.mule.api.metadata.MetadataKey;
+import org.mule.extension.api.annotation.param.metadata.Content;
+import org.mule.metadata.api.builder.BaseTypeBuilder;
+import org.mule.metadata.api.model.MetadataFormat;
+import org.mule.metadata.api.model.MetadataType;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Responsible of resolve the {@link MetadataKey}'s and and {@link MetadataType}'s of a given
+ * Operation.
+ *
+ * @since 1.0
+ */
+public interface MetadataResolver
+{
+
+    /**
+     * Resolves the list of types that can be described, representing them
+     * as a list of {@link MetadataKey}
+     *
+     * @param context MetaDataContext of the MetaData resolution
+     * @return A list of {@link MetadataKey} of the available types
+     * @throws MetadataResolvingException
+     */
+    default List<MetadataKey> getMetadataKeys(MetadataContext context) throws MetadataResolvingException
+    {
+        return Collections.emptyList();
+    }
+
+    /**
+     * Given a {@link MetadataKey} of a type, resolves their {@link MetadataType} which
+     * represents the type structure.
+     * This {@link MetadataType} will be considered as the main input of an Operation
+     * for their parameter marked as {@link Content}.
+     *
+     * @param context MetaDataContext of the MetaData resolution
+     * @param key     {@link MetadataKey} of the type which's structure has to be resolved
+     * @return the {@link MetadataType} of the {@link Content} parameter
+     * @throws MetadataResolvingException
+     */
+    default MetadataType getContentMetadata(MetadataContext context, MetadataKey key) throws MetadataResolvingException
+    {
+        return BaseTypeBuilder.create(new MetadataFormat("nullFormat", "nullFormat")).nullType().build();
```

Comment:
Having a constant in the interface would make it public, and is not something worth exposing. This "nullFormat" string is only passed because MetadataFormat does not allow for blank id's. Is not and actual id that anyone would use or compare.